engines/stark/ui/menu/locationscreen.cpp

```diff
@@ -180,6 +180,9 @@ void StaticLocationWidget::onClick() {
 		_soundMouseClick->play();
 	}
 
+	// Ensure the click sound is played completely
+	while (_soundMouseClick->isPlaying()) {};
+
```

Comment:
Based on the original game (steam version here), the click sound is completely played before the action is executed. Although using ```while``` loop here will freeze the game, but that is also what will happen in the original game, so this should be enough "original...ish".

Reply:
Wait the pointer needs to be checked beforehand.